src/include/daos_types.h

```diff
@@ -201,6 +201,9 @@ typedef struct {
 	uint64_t	hi;
 } daos_obj_id_t;
 
+/** max pool/cont attr size */
+#define DAOS_ATTR_NAME_MAX 511
```

Comment:
Could you please advise what error is reported if someone is trying to set_attr an attribute which is bigger than this size?

Reply:
currently attribute name size is not limited during pool/cont set-attr() operation, but should we report an error and only forget about this specific one or also abort all others attributes part of the same request ? Or may be it could be silently truncated to DAOS_ATTR_NAME_MAX with no error ?

Reply:
truncating is bad IMO. we should just fail if we impose this len limit.

Reply:
Limit/error handling is done in attr_check_input()